adapter/PreciceInterface.c

```diff
@@ -160,6 +160,40 @@ void Precice_WriteIterationCheckpoint(SimulationData *sim, double *v)
   memcpy(sim->coupling_init_v, v, sizeof(double) * sim->mt * sim->nk);
 }
 
+void Precice_ReadIterationCheckpointModal(SimulationData *sim, double *dofs, double *derivatives, int nev)
+{
+
+  printf("Adapter reading checkpoint...\n");
+  fflush(stdout);
+
+  // Reload time
+  *(sim->theta) = sim->coupling_init_theta;
+
+  // Reload step size
+  *(sim->dtheta) = sim->coupling_init_dtheta;
+
+  // Reload DOFs in eigenmodes space & counters
+  memcpy(dofs, sim->eigenDOFs, sizeof(double) * nev);
+  memcpy(derivatives, sim->eigenDOFsDerivatives, sizeof(double) * nev);
```

Comment:
Wouldn't it make more sense to add these in the main read/write checkpoint methods and only enable them if this is a modal simulation?

Reply:
That would probably mess up the arguments.unless i move the calculix data into the SimData, but that would be a bit messy and a break of separation of concerns in my opinion

Reply:
If you feel that this would be more complicated, we can also leave it like this and look at it again in the future.